File: ruby/shuffled_sentence.rb
s = 'gdJL3 qZVrDtrdt4 hNbOybAYKc1 IPUpNYOFYdFbP2'

def sort_sentence(s)
  s.split(' ').sort_by { |word| word[-1] }.join(' ').gsub(/\d+/, '')
end

p sort_sentence(s)

File: python/shuffled_sentence.py
s = 'gdJL3 qZVrDtrdt4 hNbOybAYKc1 IPUpNYOFYdFbP2'

def sortSentence(s):
  sentence = ' '.join(sorted(s.split(' '), key = lambda word: word[-1]))
  return ''.join([i for i in sentence if not i.isdigit()])


print(sortSentence(s))